docs/intelligentapps/finetune.md

```diff
@@ -1,317 +1,403 @@
-| `infra/provision/finetuning.bicep` | Contains templates for provisioning Azure resources for fine-tuning. |
-| `infra/finetuning.config.json` |The configuration file, generated by the `AI Toolkit: Provision Azure Container Apps job for fine-tuning` command. It is used as input for other remote command palettes. |
+The learning rate, learning rate scheduler and number of warmup steps all interact with each other. Keeping two fixed and varying the third will give you better insights into how they change the output of training on your dataset.
 
-### Configuring secrets for fine-tuning in Azure Container Apps
+The weight decay and LoRA dropout parameters are there to help to control overfitting. If you see that your adapter is not generalising well from your training set to your evaluation set, try increasing the values of these parameters.
 
-Azure Container App Secrets provide a secure way to store and manage sensitive data within Azure Container Apps, like HuggingFace tokens and Weights & Biases API keys. Using AI toolkit's command palette, you can input the secrets into the provisioned Azure container app job(as stored in `./finetuning.config.json`). These secrets are then set as **environment variables** in all containers.
+The adam_ parameters affect how the Adam optimizer behaves during the training steps. For more information on that optimizer see, for example, the Pytorch documentation.
 
-#### Steps
+Many of the other parameters exposed are analogous to their equivalently named counterparts in the PEFT library. For more information on those, see the transformers documentation.
 
-1. In the Command Palette, type and select `AI Toolkit: Add Azure Container Apps Job secret for fine-tuning`
+### Data hyperparameters
 
-    ![Add secret](./images/finetune/command-add-secret.png)
+The data hyperparameters train_nsamples and test_nsamples control how many samples to take for training and testing respectively. Using more samples from your training set is normally a good idea. Using more test samples will give you test metrics that are less noisy, but each evaluation run will take longer.
 
-1. Input Secret Name and Value: You'll be prompted to input the name and value of the secret.
-    ![Input secret name](./images/finetune/input-secret-name.png)
-    ![Input secret](./images/finetune/input-secret.png)
-    For example, if you're using private HuggingFace dataset or models that need Hugging Face access control, set your HuggingFace token as an environment variable [`HF_TOKEN`](https://huggingface.co/docs/huggingface_hub/package_reference/environment_variables#hftoken) to avoid the need for manual login on the Hugging Face Hub.
+The train_batch_size and test_batch_size parameters control how many samples should be used in each batch for training and testing, respectively. You can normally use more batches for testing than training, because running a test example takes less GPU memory than a training example.
```

Comment:
```suggestion
The `train_batch_size` and `test_batch_size` parameters control how many samples should be used in each batch for training and testing, respectively. You can normally use more batches for testing than training, because running a test example takes less GPU memory than a training example.
```